sos/plugins/collectd.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2016 Archit Sharma <archit.sh@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import re
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class Collectd(Plugin, RedHatPlugin):
+    '''
+    Collectd config collector
+    '''
+    plugin_name = "collectd"
+    profiles = ('services', 'webserver')
+
+    files = (
+        '/etc/collectd.conf',
+    )
+
+    def setup(self):
+        self.add_copy_spec([
+            '/etc/collectd.conf',
+            '/etc/collectd.d/*.conf',
+        ])
```

Comment:
I've added a new commit that uses `postproc()` to redact  `URL, Host, Port, Password, User, Address` for different plugins that might be loaded in collectd, per se, PostgreSQL, mysql, modbus, curl_xml, curl_json, ascent, amqp and so on. I've tested this against a current deployment. 

Let me know if this is too much obfuscation and if people might require Host/Port addresses that plugins connect to (and leave out just the passwords). Note that Hostname has not been redacted (it's the one that represents the hostname of the machine)